manual/manual/tutorials/coreexamples.etex

```diff
@@ -537,6 +548,39 @@ let temporarily_set_reference ref newval funct =
     raise x;;
 \end{caml_example}
 
+An alternative to "try"\ldots"with" is to catch the exception while
+pattern matching:
+\begin{caml_example}{toplevel}
+let assoc_opt x l =
+  match List.assoc x l with
+  | exception Not_found -> None
+  | x -> Some x;;
+\end{caml_example}
+Note that this construction is only useful if the exception is raised
+between "match"\ldots"with". Moreover, exceptions can only
+appear at the toplevel of such a pattern match.
+For instance, exceptions cannot be combined with an or-pattern:
+"exception A | exception B ->"\ldots .
```

Comment:
Indeed, exception *clauses* can't be combined with an `or-pattern`.  I think this is just a limitation of the current implementation, which could perhaps be indicated by writing "... cannot currently be combined ..."

However, the exception *patterns* themselves can be combined with an or-pattern: `exception (A|B) -> `

Reply:
The term `clause` is not used currently inside the tutorial, thus I think that `exception *cases*` would be clearer to the reader. But I agree that is better to avoid giving the impression that `exception (A|B)` is not valid.